design/expected-hardware-configuration-validation.md

```diff
@@ -0,0 +1,138 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# Hardware Classification Controller
+
+## Table of Contents
+
+<!--ts-->
+   * [Hardware Classification](#title)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+      * [Proposal](#proposal)
+         * [Implementation Details/Notes/Constraints ](#implementation-detailsnotesconstraints-optional)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Work Items](#work-items)
+         * [Dependencies](#dependencies)
+         * [Test Plan](#test-plan)
+      * [References](#references)
+
+## Summary
+
+Code with the ability to perform the comparison on inspected and ready hosts and update the host with labels containing matched profiles.
+
+## Motivation
+
+We are closely working with Airship team, There is one requirement in Airship for classification.
+As an User, I want to validate and classify the hosts based on my hardware requirement so that I will get to know how many hosts match to my profile and accordingly I can go ahead and provision these hosts only.
+This feature helps user to avoid runtime failures and also increases the performances for workload deployments.
+
+### Goals
+
+Automatically classify bare metal hosts based on user-provided rules in a way that other controllers can use, without having to evaluate all of those rules themselves.
+
+## Proposal
+
+We are creating proposal based on the discussion with metal3 community on issue #351:
+https://github.com/metal3-io/baremetal-operator/issues/351
+
+We compared introspection data of Ironic with metal3 schema and we found that default  hardware
+configuration needs to be added by introducing a new CRD `HardwareClassification`.
+
+We will write a CRD that checks inspected baremetal hosts against hardware configuration and add label as "MATCHES" for matched profile into baremetal host.
+  
+### Implementation Details/Notes/Constraints
+
+Link for Existing Metal3 Specs
+Please refer metal3 spec for bare-metal:
+https://github.com/metal3-io/baremetal-operator/blob/master/deploy/crds/metal3.io_baremetalhosts_crd.yaml
+
+* Below is sample yaml of Kind HardwareClassification.
+    
+   ```yaml
+   
+   apiVersion: metal3.io.sigs.k8s.io/v1alpha1
+   kind: HardwareClassification
+   metadata:
+     name: profile1
+   spec:
+   expectedValidationConfiguration:
+      namespace: metal3
+      CPU:
+         minimumCount: 4
+         minimumSpeed: 4.3
+      Disk:
+         minimumSizeGB: 2200
+         minimumCount: 2
+      NICS:
```

Comment:
This is the only example that is plural. Maybe just `NIC`?

Reply:
ok